src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> currentNode = tail;
+        tail = newNode;
+        tail.prev = currentNode;
+        currentNode.next = tail;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new IndexOutOfBoundsException("Can't operate with the value "
+                    + "index is out of bounds");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
+        addBefore(value, node);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        unLink(node);
+        return node.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (object == null
+                    ? node.item == null
+                    : object.equals(node.item)) {
+                unLink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size()) {
```

Comment:
You have access to this variable. Better not method 
```suggestion
        if (index < 0 || index >= size) {
```